ns/init_linux.go

```diff
@@ -100,12 +100,7 @@ func getSupportedNlFamilies() []int {
 }
 
 func loadXfrmModules() error {
-	if out, err := exec.Command("modprobe", "-va", "xfrm_user").CombinedOutput(); err != nil {
-		return fmt.Errorf("Running modprobe xfrm_user failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
-	}
-	if out, err := exec.Command("modprobe", "-va", "xfrm_algo").CombinedOutput(); err != nil {
-		return fmt.Errorf("Running modprobe xfrm_algo failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
-	}
+	// Those are automatically loaded when someone opens the socket anyway.
```

Comment:
This function is not exported, so perhaps it should be removed altogether; looks like `checkXfrmSocket()` is also no longer called now, so could be removed (do we need to try that socket to force the modules to load?)

perhaps add a comment in the calling function about these being automatically loaded, e.g.;

```patch
diff --git a/ns/init_linux.go b/ns/init_linux.go
index 6a56b837..ed78bfa6 100644
--- a/ns/init_linux.go
+++ b/ns/init_linux.go
@@ -74,17 +74,10 @@ func NlHandle() *netlink.Handle {
 }
 
 func getSupportedNlFamilies() []int {
-       fams := []int{syscall.NETLINK_ROUTE}
-       // NETLINK_XFRM test
-       if err := loadXfrmModules(); err != nil {
-               if checkXfrmSocket() != nil {
-                       logrus.Warnf("Could not load necessary modules for IPSEC rules: %v", err)
-               } else {
-                       fams = append(fams, syscall.NETLINK_XFRM)
-               }
-       } else {
-               fams = append(fams, syscall.NETLINK_XFRM)
-       }
+       // No need to "modprobe" xfrm_user and xfrm_algo, as they are loaded
+       // automatically when a socket is opened
+       fams := []int{syscall.NETLINK_ROUTE, syscall.NETLINK_XFRM}
+
        // NETLINK_NETFILTER test
        if err := loadNfConntrackModules(); err != nil {
                if checkNfSocket() != nil {
@@ -99,21 +92,6 @@ func getSupportedNlFamilies() []int {
        return fams
 }
 
-func loadXfrmModules() error {
-       // Those are automatically loaded when someone opens the socket anyway.
-       return nil
-}
-
-// API check on required xfrm modules (xfrm_user, xfrm_algo)
-func checkXfrmSocket() error {
-       fd, err := syscall.Socket(syscall.AF_NETLINK, syscall.SOCK_RAW, syscall.NETLINK_XFRM)
-       if err != nil {
-               return err
-       }
-       syscall.Close(fd)
-       return nil
-}
-
 func loadNfConntrackModules() error {
        if out, err := exec.Command("modprobe", "-va", "nf_conntrack").CombinedOutput(); err != nil {
                return fmt.Errorf("Running modprobe nf_conntrack failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
```

Reply:
Is XFRM optional? My changeset took care that it still kept the checkXfrmSocket functionality (including the case where it returns with error) in case the entire thing is supposed to be optional.

Your proposed change would make fams always contain something, right? Then it wouldn't be optional.

Or is it supposed to be required anyway?

Reply:
Right. But is it supposed to be optional? maybe checkXfrmSocket was supposed to be called still.

Frankly, I hadn't seen the part

```
        if err := loadXfrmModules(); err != nil {
```

Maybe it would be better to call checkXfrmSocket anyway--and warn on error.

Reply:
Yes, was suspecting the same, hence my question above;

> do we need to try that socket to force the modules to load?

So probably call it unconditionally, and indeed produce an error on failure

Reply:
I've updated the branch to do that.